plugins/processors/ec2tagger/constants.go

```diff
@@ -0,0 +1,79 @@
+  ## subset of ec2 tags), ec2tagger will retry every 3 minutes until all the tags/volumes (as specified by
+  ## "ec2_instance_tag_keys"/"ebs_device_keys") are retrieved successfully. (Note when the specified list is ["*"],
+  ## there is no way to check if all tags/volumes are retrieved, so there is no retry in that case)
+  # refresh_interval_seconds = 60
+  ##
+  ## Add tags for EC2 Metadata fields.
+  ## Supported fields are: "InstanceId", "ImageId" (aka AMI), "InstanceType"
+  ## If the configuration is not provided or it has an empty list, no EC2 Metadata tags are applied.
+  # ec2_metadata_tags = ["InstanceId", "ImageId", "InstanceType"]
+  ##
+  ## Add tags retrieved from the EC2 Instance Tags associated with this instance.
+  ## If this configuration is not provided, or has an empty list, no EC2 Instance Tags are applied.
+  ## If this configuration contains one entry and its value is "*", then ALL EC2 Instance Tags for the instance are applied.
+  ## Note: This plugin renames the "aws:autoscaling:groupName" EC2 Instance Tag key to be spelled "AutoScalingGroupName".
+  ## This aligns it with the AutoScaling dimension-name seen in AWS CloudWatch.
+  # ec2_instance_tag_keys = ["aws:autoscaling:groupName", "Name"]
+  ##
+  ## Retrieve ebs_volume_id for the specified devices, add ebs_volume_id as tag. The specified devices are
+  ## the values corresponding to the tag key "disk_device_tag_key" in the input metric.
+  ## If this configuration is not provided, or has an empty list, no ebs volume is applied.
+  ## If this configuration contains one entry and its value is "*", then all ebs volume for the instance are applied.
+  # ebs_device_keys = ["/dev/xvda", "/dev/nvme0n1"]
+  ##
+  ## Specify which tag to use to get the specified disk device name from input Metric
+  # disk_device_tag_key = "device"
+  ##
+  ## Amazon Credentials
+  ## Credentials are loaded in the following order
+  ## 1) Assumed credentials via STS if role_arn is specified
+  ## 2) explicit credentials from 'access_key' and 'secret_key'
+  ## 3) shared profile from 'profile'
+  ## 4) environment variables
+  ## 5) shared credentials file
+  ## 6) EC2 Instance Profile
+  # access_key = ""
+  # secret_key = ""
+  # token = ""
+  # role_arn = ""
+  # profile = ""
+  # shared_credential_file = ""
+`
+
+const (
+	ec2InstanceTagKeyASG = "aws:autoscaling:groupName"
+	cwDimensionASG       = "AutoScalingGroupName"
+	mdKeyInstanceId      = "InstanceId"
+	mdKeyImageId         = "ImageId"
+	mdKeyInstanceType    = "InstanceType"
+	ebsVolumeId          = "EBSVolumeId"
+)
+
+const (
+	allowedIMDSRetries = 2
+)
+
+var (
+	defaultIMDSTimeout     = 1 * time.Second
+	defaultRefreshInterval = 180 * time.Second
+	backoffSleepArray      = []time.Duration{0, 1 * time.Minute, 1 * time.Minute, 3 * time.Minute, 3 * time.Minute, 3 * time.Minute, 10 * time.Minute} // backoff retry for ec2 describe instances API call. Assuming the throttle limit is 20 per second. 10 mins allow 12000 API calls.
```

Comment:
This should be a random number imo. Think all systems send out the call at the same time. Then next round they all send at the same time. It needs to be random within an interval. This comment is probs out of scope for this pr but something to think about for best behaved agents. @straussb thoughts.